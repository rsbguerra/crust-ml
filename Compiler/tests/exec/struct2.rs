struct S {
    a: i32, b: bool
}

fn main() {
    let s = S { a: 42, b: true };
    if s.a == 42 { print!("yes!\n") } else { print!("oups!\n") }
    if s.b       { print!("true!\n") } else { print!("false!\n") }
    let mut t = S { a: 0, b: false };
    if t.a == 0 { print!("yes!\n") } else { print!("oups!\n") }
    if t.b       { print!("true!\n") } else { print!("false!\n") }
    t = s;
    if t.a == 42 { print!("yes!\n") } else { print!("oups!\n") }
    if t.b       { print!("true!\n") } else { print!("false!\n") }
}
