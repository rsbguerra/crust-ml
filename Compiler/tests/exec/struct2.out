yes!
true!
yes!
false!
yes!
true!
